Code/Framework/AzCore/AzCore/Math/Ray.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/Ray.h>
+
+namespace O3DE
+{
+    Ray::Ray(const AZ::Vector3& origin, const AZ::Vector3& direction)
+        : m_origin(origin)
+        , m_direction(direction)
+    {
+        AZ_MATH_ASSERT(m_direction.IsNormalized(), "This normal is not normalized");
```

Comment:
Why can't we normalize the direction for the user?
If it fails because the direction has a magnitude too close to 0, then we can assert in that case.

Reply:
its consistent with some other use cases were we assume the user would provide a normalized vector.